java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -711,10 +711,51 @@ class DatabaseImpl(
         }
     }
 
+    /*
+     * Removes all refs (in entity_refs table) that are not being used.
+     */
+    private fun removeUnusedRefs(db: SQLiteDatabase) {
+        db.transaction {
+            // Find all refs used in singleton fields.
+            val singletonFieldRefs = rawQuery(
+                """
+                    SELECT field_values.value_id
+                    FROM field_values
+                    LEFT JOIN fields ON field_values.field_id = fields.id
```

Comment:
This could/should probably be an inner join rather than a left join - unless you also want to find any field_values for which there are no corresponding fields.... (might make sense as a sanity-check?)

Reply:
changed to inner. it should not make a difference since we are also filtering by fields.type_id?